typhon/objects/processes.py

```diff
@@ -57,17 +64,76 @@ def recv(self, atom, args):
 
 
 @autohelp
+class ProcessExitInformation(Object):
+    """
+    Holds a process' exitStatus and terminationSignal
+    """
+
+    def __init__(self, exitStatus, terminationSignal):
+        self.exitStatus = exitStatus
+        self.terminationSignal = terminationSignal
+
+    def toString(self):
+        return (u'<ProcessExitInformation exitStatus=%d,'
+                u' terminationSignal=%d>' % (self.exitStatus,
+                                             self.terminationSignal))
+
+    def recv(self, atom, args):
+        if atom is EXITSTATUS_0:
+            return IntObject(self.exitStatus)
+
+        if atom is TERMINATIONSIGNAL_0:
+            return IntObject(self.terminationSignal)
+
+        raise Refused(self, atom, args)
+
+
+@autohelp
 class SubProcess(Object):
     """
     A subordinate process of the current process, on the local node.
     """
+    EMPTY_PID = -1
+    EMPTY_EXIT_AND_SIGNAL = (-1, -1)
 
-    def __init__(self, pid, argv, env):
-        self.pid = pid
+    def __init__(self, vat, process, argv, env):
+        self.pid = self.EMPTY_PID
```

Comment:
I anticipate a state machine being a good thing here, but I don't have a turnkey state machine solution for RPython yet. A similar state machine solution would be good in ``typhon.objects.files``.

No action needed, just irritated that I haven't solved this problem.